rule/fetcher_default.go

```diff
@@ -190,31 +187,26 @@
 }
 
 func (f *FetcherDefault) watch(ctx context.Context, watcher *fsnotify.Watcher, events chan event) error {
-	var pc map[string]interface{}
-
-	viperx.AddWatcher(func(e fsnotify.Event) error {
-		if reflect.DeepEqual(pc, viper.Get(configuration.ViperKeyAccessRuleRepositories)) {
-			f.r.Logger().
-				Debug("Not reloading access rule repositories because configuration value has not changed.")
-			return nil
+	f.config.AddWatcher(func(e watcherx.Event, err error) {
+		if err != nil {
+			return
 		}
+		f.registry.Logger().
+			WithField("event", e).
+			Debugf("Reloading access rule repositories because configuration value has changed.")
 
 		f.enqueueEvent(events, event{et: eventRepositoryConfigChanged, source: "viper_watcher"})
-
-		return nil
 	})
 	f.enqueueEvent(events, event{et: eventRepositoryConfigChanged, source: "entrypoint"})
 
-	var strategy map[string]interface{}
-	viperx.AddWatcher(func(e fsnotify.Event) error {
-		if reflect.DeepEqual(strategy, viper.Get(configuration.ViperKeyAccessRuleMatchingStrategy)) {
-			f.r.Logger().
-				Debug("Not reloading access rule matching strategy because configuration value has not changed.")
-			return nil
+	f.config.AddWatcher(func(e watcherx.Event, err error) {
+		if err != nil {
+			return
 		}
-
+		f.registry.Logger().
+			WithField("event", e).
+			Debugf("Reloading strategy because configuration value has changed.")
 		f.enqueueEvent(events, event{et: eventMatchingStrategyChanged, source: "viper_watcher"})
```

Comment:
Similar question here. We could probably just have one watcher in that case and submit both events at once if we don't check whether the value has actually changed?

Reply:
Yes, I'll refactor